[ADD] edi: Improve document status visibility

armandcela opened the pull request:
Add a new field to show edi document process status to have more visibility so that user understand easier at which state the edi document is in. SE-1104
Signed-off-by: Armand Cela <armand.cela@unipart.io>

rajeeshknambiar commented on addons/edi/models/edi_document.py:
```diff
@@ -169,6 +176,18 @@ class EdiDocument(models.Model):
+            ("waiting", "Waiting"),
+            ("preparing", "Preparing"),
+            ("executing", "Executing")
+        ],
```
Would it be good to set default `processing_state` as `waiting`?

rajeeshknambiar replied:
Ah, line 366 expects default processing_state to be none/empty, so ignore this comment.

armandcela replied:
I dont think so as in generall will be empty on create, it will be waiting only when is created from transfer and allow_process config is not set True on transfer

rajeeshknambiar commented on addons/edi/models/edi_document.py:
```diff
@@ -99,7 +99,7 @@ def _default_project_id(self):
 
     @api.model
-    def autocreate(self, inputs):
+    def autocreate(self, inputs, allow_process=True):
```
Nit: may be name the parameter as `auto_process`? (Feel free to ignore if you prefer the existing name).

armandcela replied:
The parameter name is same with variable name on gateway class i think

rajeeshknambiar commented on addons/edi/models/edi_transfer.py:
```diff
@@ -163,13 +164,22 @@ def send_outputs(self, conn):
 
     def do_transfer(self, conn):
-        """Receive input attachments, process documents, send outputs"""
+        """Receive input attachments, send outputs"""
```
`send outputs if flag is set` ?

armandcela replied:
hmm there was not if flag is set before, did i remove it?

rajeeshknambiar commented on addons/edi/tests/common.py:
```diff
@@ -46,6 +46,11 @@ class EdiCase(common.SavepointCase):
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
+        patch_cr = patch.object(
```
Please add a comment about the significance of patching (commit in the middle of savepoint etc.) so that will help you two weeks later :smiley:

armandcela replied:
I forgot already tbh, your memory is quite good :)

rajeeshknambiar commented on addons/edi/views/edi_document_views.xml:
```diff
@@ -62,8 +62,11 @@
 	      <label for="name" class="oe_edit_only"/>
 	      <h1>
-		<field name="name" placeholder="e.g. PMD00001"/>
+					<field name="name" placeholder="e.g. PMD00001"/>
```
Nit: spacing looks off here.

armandcela replied:
TBH the indentation is not well on original file, there are spaces and tabs mixed in original file

rajeeshknambiar commented on addons/edi/views/edi_document_views.xml:
```diff
@@ -108,6 +111,7 @@
 	  <field name="gateway_id" groups="edi.group_edi_gateway_view"/>
 	  <field name="transfer_id" groups="edi.group_edi_gateway_view"/>
 	  <field name="state"/>
+		<field name="processing_state" optional="hide"/>
```
Nit: spacing/indenting is off, in changes below also.

armandcela replied:
TBH the indentation is not well on original file, there are spaces and tabs mixed in original file

armandcela committed:
[ADD] edi: Adding more visibility of intermediary statuses of EDI document

Adding more visibiility of edi document when the document is preparing and executing
SE-1104
Signed-off-by: Armand Cela <armand.cela@unipart.io>

armandcela committed:
[ADD] edi: Patching cursor not to commit changes on unit tests

SE-1104
Signed-off-by: Armand Cela <armand.cela@unipart.io>
